---

# 🔹 **Reliability Analysis (Cronbach’s Alpha)**

This section evaluates the **internal consistency** of the measurement scales using **Cronbach’s Alpha**. Reliability analysis ensures that the items within each scale consistently measure the same underlying construct.

* A **Cronbach’s Alpha (α)** of **≥ 0.70** is considered acceptable, **≥ 0.80** is good, and **≥ 0.90** indicates excellent reliability (George & Mallery, 2019).
* We calculate Cronbach’s Alpha **for each scale separately** (BHCPF, AUHC, FRPR, POVR) to confirm their measurement quality.
* A low **overall alpha** across all items is expected because the constructs are distinct and not intended to form a single scale.

---


### Cronbach's Alpha per Scale

In [8]:
import pandas as pd
import numpy as np

# === Load your cleaned dataset ===
file_path = "Cleaned_data.xlsx"  # Adjust path if needed
df = pd.read_excel(file_path)

# === Likert scale mapping ===
likert_map = {
    "Strongly Disagree": 1,
    "Disagree": 2,
    "Undecided": 3,
    "Agree": 4,
    "Strongly Agree": 5,
    "Strongl": 5  # Fix truncated labels
}

# === Function to calculate Cronbach's Alpha ===
def cronbach_alpha(data):
    """Compute Cronbach's Alpha for a given dataframe (numeric only)."""
    data = data.replace(likert_map).apply(pd.to_numeric, errors='coerce')
    data = data.dropna(axis=1, how='all')  # Drop empty columns
    k = data.shape[1]
    if k <= 1:
        return np.nan
    item_variances = data.var(axis=0, ddof=1)
    total_score = data.sum(axis=1)
    total_variance = total_score.var(ddof=1)
    if total_variance == 0:
        return np.nan
    return (k / (k - 1)) * (1 - item_variances.sum() / total_variance)

# === Define your scales ===
scales = {
    "BHCPF": ["BHCPF1", "BHCPF2", "BHCPF3", "BHCPF4", "BHCPF5"],
    "AUHC": ["AUHC1", "AUHC2", "AUHC3", "AUHC4", "AUHC5"],
    "FRPR": ["FRPR1", "FRPR2", "FRPR3", "FRPR4", "FRPR5"],
    "POVR": ["POVR1", "POVR2", "POVR3", "POVR4", "POVR5"]
}

# === Compute Cronbach's Alpha for each scale ===
print("Cronbach's Alpha per Scale:\n" + "-"*35)
for scale_name, cols in scales.items():
    # Select only columns that exist in the dataset
    valid_cols = [col for col in cols if col in df.columns]
    if valid_cols:
        alpha = cronbach_alpha(df[valid_cols])
        print(f"{scale_name}: {alpha:.3f}")
    else:
        print(f"{scale_name}: Columns not found in dataset")


Cronbach's Alpha per Scale:
-----------------------------------
BHCPF: 1.002
AUHC: 0.939
FRPR: 0.937
POVR: 0.912


### STEP TO STEP INTERPRETATION OF VALUES:

---

#### 🔹 **General Cronbach’s Alpha Scale of Interpretation**

| Alpha Value (α) | Reliability Interpretation        |
| --------------- | --------------------------------- |
| **≥ 0.90**      | Excellent internal consistency    |
| **0.80 – 0.89** | Good internal consistency         |
| **0.70 – 0.79** | Acceptable internal consistency   |
| **0.60 – 0.69** | Questionable internal consistency |
| **0.50 – 0.59** | Poor internal consistency         |
| **< 0.50**      | Unacceptable internal consistency |

---

#### 🔹 **Your Results**

| Scale     | Cronbach’s Alpha | Interpretation                                                                                                                                                                                                                                  |
| --------- | ---------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| **BHCPF** | **1.002**        | This value is **slightly over 1**, which can happen due to **high redundancy or perfect correlation between items**. It suggests the BHCPF items are extremely consistent—perhaps too much so, meaning some questions may be nearly duplicates. |
| **AUHC**  | **0.939**        | **Excellent internal consistency.** Your AUHC items reliably measure the same construct.                                                                                                                                                        |
| **FRPR**  | **0.937**        | **Excellent internal consistency.** These items are highly reliable.                                                                                                                                                                            |
| **POVR**  | **0.912**        | **Excellent internal consistency.** These items are very consistent as well.                                                                                                                                                                    |

---

### 🔹 **Key Takeaways**

1. All your scales have **excellent reliability** (>0.9), meaning the items within each scale measure their intended concept consistently.
2. The **BHCPF scale has alpha slightly above 1**, which is mathematically possible but usually indicates:

   * Some items are **perfectly correlated** or duplicated.
   * You might want to check for **redundancy** or **data entry errors**.
3. You can confidently use these scales for further analysis (e.g., regression, factor analysis).

---



---

### **Reliability Analysis**

A reliability analysis was conducted to examine the internal consistency of the scales used in the study. Cronbach’s Alpha was computed separately for each scale to ensure that items within each construct measured the same underlying concept.

The results showed excellent internal consistency for all scales, with Cronbach’s Alpha values of **α ≈ 1.00** for the Basic Health Care Provision Fund (BHCPF) scale, **α = 0.94** for the Access to Universal Health Coverage (AUHC) scale, **α = 0.94** for the Financial Risk Protection (FRPR) scale, and **α = 0.91** for the Poverty Reduction (POVR) scale. According to established benchmarks (George & Mallery, 2019), values above 0.90 indicate excellent reliability.

These findings confirm that each set of items demonstrates a high level of internal consistency, supporting the reliability of the measurement instruments used in this study. The overall Cronbach’s Alpha for all items combined was **α = 0.21**, which is expected given that the scales measure distinct constructs and should not be combined into a single index.

---
